udiskie: 1.4.8 -> 1.5.1

rycee opened the pull request:
###### Motivation for this change

Version bump and use of GApps wrapper and icon hooks. Unfortunately, despite the use of gapps wrapper I still get some missing icons in the tray menu.
###### Things done
- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

rycee committed:
udiskie: use wrap GApps and icon setup hooks

rycee committed:
udiskie: 1.4.8 -> 1.5.1

Also switch to `fetchFromGitHub` and do minor formatting.

rycee commented:
CC @AndersonTorres

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -27121,13 +27121,15 @@ in modules // {
 
-  udiskie = buildPythonPackage rec {
-    version = "1.4.8";
+  udiskie = buildPythonApplication rec {
```
this is indeed an application. As such, this expression should be moved outside of `python-packages.nix` and put somewhere else in `nixpkgs`, since `python-packages.nix` is only intended for libraries or packages that need to be available for Python versions.

rycee replied:
Added a commit that moves the package to `applications/misc/udiskie`. I kept a reference in `pythonPackages` for backwards compatibility. I'm not sure whether that is necessary, though…

rycee committed:
udiskie: move to all-packages

This is an application, not a python library and should therefore be in
its own package.

FRidh commented on pkgs/applications/misc/udiskie/default.nix:
```diff
@@ -0,0 +1,47 @@
+  };
+
+  preConfigure = ''
+    export XDG_RUNTIME_DIR=/tmp
```
you can set environment variables outside of the phases. Just passing `XDG_RUNTIME_DIR=/tmp;` is sufficient.

bjornfor replied:
> you can set environment variables outside of the phases. Just passing XDG_RUNTIME_DIR=/tmp; is sufficient.

I know this feature is used all over the place, but I always felt that was weird/wrong. Like a leaky abstraction.

rycee replied:
As far as I can tell the `XDG_RUNTIME_DIR` variable doesn't matter at all in the udiskie build. The package builds and runs just fine without it. I removed it and force-pushed.

FRidh replied:
@bjornfor could you explain why you think it is weird or wrong? I can imagine it is strange to set something outside of a phase, and so it might not be clear when exactly these env vars are set.

bjornfor replied:
On 12 September 2016 at 11:07, Frederik Rietdijk
<notifications@github.com> wrote:
>
> In pkgs/applications/misc/udiskie/default.nix:
>
> > +, gobjectIntrospection, gtk3, hicolor_icon_theme, libnotify
> > +, pythonPackages, udisks2, wrapGAppsHook }:
> > +
> > +pythonPackages.buildPythonApplication rec {
> > +  name = "udiskie-${version}";
> > +  version = "1.5.1";
> > +
> > +  src = fetchFromGitHub {
> > +    owner = "coldfix";
> > +    repo = "udiskie";
> > +    rev = version;
> > +    sha256 = "01x5fvllb262x6r3547l23z7p6hr7ddz034bkhmj2cqmf83sxwxd";
> > +  };
> > +
> > +  preConfigure = ''
> > +    export XDG_RUNTIME_DIR=/tmp
>
> @bjornfor could you explain why you think it is weird or wrong? I can imagine it is strange to set something outside of a phase, and so it might not be clear when exactly these env vars are set.

Why I think it's weird:
* No clear distinction between Nix value and Shell value. It's just
"all the same".
* A very "global" scope. No way to set an envvar for _just_ configure phase.

I think something like this would be cleaner (inspired by Buildroot):

    builderEnv = { MYVAR = "VALUE"; };    # this will be set for all
processes spawned in a build
    configureEnv = { MYVAR = "VALUE"; };   # just when invoking ./configure

I guess it all comes down to ones perception of the
stdenv.mkDerivation function. Mine is probably wrong :-)

FRidh replied:
Yes I agree that would be much nicer. We should build wrappers in such a way as well I think.

But for the case here, since at this point the state is inherited from one phase to another, there is no difference between using `export X=y;` or `X=y;` except if you want to export the var after a certain phase.

rycee commented:
Thanks for the review!
